File: app/graphql/mutations/create_password_reset_mutation.rb
Mutations::CreatePasswordResetMutation = GraphQL::Relay::Mutation.define do
  name  'CreatePasswordReset'

  input_field :email, !types.String

  return_field :user, Types::UserType

  resolve ->(object, inputs, ctx) {
    create_password_reset = API::CreatePasswordReset.call(email: inputs[:email])

    if create_password_reset.success?
      { user: nil }
    else
      GraphQL::ExecutionError.new(create_password_reset.error)
    end
  }
end

File: client/mutation/CreatePasswordResetMutation.js
import { commitMutation, graphql } from 'react-relay'

const mutation = graphql`
  mutation CreatePasswordResetMutation($input: CreatePasswordResetInput!) {
    createPasswordReset(input: $input) {
      user {
        id
      }
    }
  }
`

function commit({ environment, input, onCompleted, onError }) {
  const variables = { input }

  commitMutation(environment, {
    mutation,
    variables,
    onCompleted,
    onError,
  })
}

export default {
  commit,
}
